### PR TITLE
Fix: several typos in the documentation

### PR DESCRIPTION
**Pull request by tudorpintea999**
### Description of the Changes

Hello
I fixed some typos in the docs,
Hope it helps.
Br,
Tudor.

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/929)

### CODE REVIEW

File: components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/contract-abi.adoc [hunk 1]
```diff
@@ -621,7 +621,7 @@ As `TestEnum`, `Var1` and `Var2` are of kind `nested`, a selector should be adde
 
 === Flattened enum variants
 
-You may not want to nest enums when serializing the event. For example, if you write an ERC20 as a component (not a contract) which is pluggable anywhere, you may not want the contract to modify the keys of known events such as `Transfer`.
+You may not want to nest enums when serializing the event. For example, if you write an ERC20 as a component (not a contract) that is pluggable anywhere, you may not want the contract to modify the keys of known events such as `Transfer`.
```
Comment (stoobie) [inline components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/contract-abi.adoc:624]:
```suggestion
You may not want to nest enums when serializing the event. For example, if you write an ERC-20 as a component, not a contract, that is pluggable anywhere, you might not want the contract to modify the keys of known events such as `Transfer`.
```

File: components/Starknet/modules/cli/pages/starkli.adoc [hunk 2]
```diff
@@ -70,7 +70,7 @@ When working with the CLI, you can manually set the endpoints for the gateways t
 interact with Starknet, by including the following options:
 
 `--feeder_gateway_url`:: Sets the custom endpoint for read commands.
-`--gateway_url`:: Sets the custom endpoint for write commands.
+`--gateway_url`:: Sets the custom endpoint for writing commands.
```
Comment (stoobie) [inline components/Starknet/modules/cli/pages/starkli.adoc:73]: This change is incorrect. The sentence is talking about read/write commands.

File: components/Starknet/modules/cli/pages/starkli.adoc [hunk 3]
```diff
@@ -387,4 +387,4 @@ The possible statuses of a transaction are:
 * `ACCEPTED_ON_L2`
 * `ACCEPTED_ON_L1`
 
-Refer to xref:architecture_and_concepts:Network_Architecture/transaction-life-cycle.adoc[this] section for more information about the transaction lifecycle.
\ No newline at end of file
+Refer to xref:architecture_and_concepts:Network_Architecture/transaction-life-cycle.adoc[this] section for more information about the transaction lifecycle.
```
Comment (stoobie) [inline components/Starknet/modules/cli/pages/starkli.adoc:390]:
```suggestion
For more information, see xref:architecture_and_concepts:Network_Architecture/transaction-life-cycle.adoc[Transaction lifecycle].
```

File: components/Starknet/modules/quick_start/pages/declare_a_smart_contract.adoc [hunk 4]
```diff
@@ -12,7 +12,7 @@ starkli --version
 scarb --version
 ----
 
-If either of the above commands fail, please visit xref:environment_setup.adoc[Setting up your environment].
+If either of the above commands fails, please visit xref:environment_setup.adoc[Setting up your environment].
```
Comment (stoobie) [inline components/Starknet/modules/quick_start/pages/declare_a_smart_contract.adoc:15]:
```suggestion
If either of the above commands fails, see xref:environment_setup.adoc[Setting up your environment].
```

File: contributing_to_docs/doc_guidelines.adoc [hunk 5]
```diff
@@ -511,7 +511,7 @@ $ <regular_user_permission_command_line_input>
 ----
 root@17617744386d:/app# ./player.py
 ----
-* Avoid using markup in code block. If you must use any markup in code blocks, see the Asciidoctor documentation on source blocks and substitutions:
+* Avoid using markup in the code block. If you must use any markup in code blocks, see the Asciidoctor documentation on source blocks and substitutions:
```
Comment (stoobie) [inline contributing_to_docs/doc_guidelines.adoc:514]:
```suggestion
* Avoid using markup in a code block. If you must use any markup in code blocks, see the Asciidoctor documentation on source blocks and substitutions:
```

**Review by stoobie — CHANGES_REQUESTED** [review]
Great comments! Thank you so much :)

Please correct the few things I mentioned, and then it's good to go.

If you prefer that I take care of it, just let me know.

**Comment by tudorpintea999** [comment]
hello @stoobie i didn t see the notification 3 days ago, check now please. Thanks !

File: components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/contract-abi.adoc [hunk 6]
```diff
@@ -621,7 +621,7 @@ As `TestEnum`, `Var1` and `Var2` are of kind `nested`, a selector should be adde
 
 === Flattened enum variants
 
-You may not want to nest enums when serializing the event. For example, if you write an ERC20 as a component (not a contract) which is pluggable anywhere, you may not want the contract to modify the keys of known events such as `Transfer`.
+You may not want to nest enums when serializing the event. For example, if you write an ERC-20 as a component, not a contract, that is pluggable anywhere, you might not want the contract to modify the keys of known events such as `Transfer`.
```
Comment (stoobie) [inline components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/contract-abi.adoc:624]:
```suggestion
You might not want to nest enums when serializing the event. For example, if you write an ERC-20 as a component, not a contract, that is pluggable anywhere, you might not want the contract to modify the keys of known events such as `Transfer`.
```

**Comment by stoobie** [comment]
Merged to new branch to generate preview before merging to dev.

**Pull request merged**